src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationCodeFixProvider.cs

```diff
@@ -93,10 +94,41 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         }
 
         private SyntaxNode CreateConditionalAccessExpression(
-            ISyntaxFactsService syntaxFacts, SyntaxGenerator generator, 
+            ISyntaxFactsService syntaxFacts, SyntaxGenerator generator, bool whenPartIsNullable,
             SyntaxNode whenPart, SyntaxNode match, SyntaxNode currentConditional)
         {
-            var memberAccess = match.Parent as TMemberAccessExpression;
+            if (whenPartIsNullable)
+            {
+                if (match.Parent is TMemberAccessExpression memberAccess)
+                {
+                    var nameNode = syntaxFacts.GetNameOfMemberAccessExpression(memberAccess);
+                    syntaxFacts.GetNameAndArityOfSimpleName(nameNode, out var name, out var arity);
+                    var comparer = syntaxFacts.IsCaseSensitive
+                        ? StringComparer.Ordinal
+                        : CaseInsensitiveComparison.Comparer;
+                    if (arity == 0 && comparer.Equals(name, nameof(Nullable<int>.Value)))
```

Comment:
💭 I'm always conflicted on cases like this. I know the semantic model is safer, but this is likely faster (which we know matters). It would be nice if `ISyntaxFactsService` provided the identifier comparer.